src/components/structures/auth/Registration.tsx

```diff
@@ -39,6 +39,14 @@ import Spinner from "../../views/elements/Spinner";
 import { AuthHeaderDisplay } from './header/AuthHeaderDisplay';
 import { AuthHeaderProvider } from './header/AuthHeaderProvider';
 
+const DEBUG = false;
+let debuglog = function(msg: string, ...args: any[]) {};
+
+if (DEBUG) {
+    // using bind means that we get to keep useful line numbers in the console
+    debuglog = logger.log.bind(console);
+}
```

Comment:
Any rationalization for why this bit is needed?

Reply:
Also, it created 3 code smells in Sonar

Reply:
It’s our standard solution for debug logging across the entire codebase, it’s used in countless files. I didn’t invent this, but while I don’t specifically like it, I didn’t want to experiment with debug logging as well (this ticket has enough things it could negatively affect already)

Reply:
Ftr its the older standard, not much new code uses it
One new approach is
https://github.com/matrix-org/matrix-react-sdk/blob/46ba14219c7d7f40120784bb8324106a1c44cc35/src/components/structures/ScrollPanel.tsx#L38-L42